src/presentation/http/router/upload.ts

```diff
@@ -0,0 +1,113 @@
+import type { FileTypes } from '@domain/entities/file';
+import type { NotePublicId } from '@domain/entities/note.js';
+import type FileUploaderService from '@domain/service/fileUploader.service.js';
+import type { MultipartFile, MultipartValue } from '@fastify/multipart';
+import type { FastifyPluginCallback } from 'fastify';
+import useNoteResolver from '../middlewares/note/useNoteResolver.js';
+import type NoteService from '@domain/service/note.js';
+
+/**
+ * Interface for upload router options
+ */
+interface UploadRouterOptions {
+  /**
+   * File uploader service
+   */
+  fileUploaderService: FileUploaderService;
+
+  /**
+   * Note service instance
+   */
+  noteService: NoteService;
+}
+
+/**
+ * Interface for upload file body
+ */
+interface UploadFileBody {
+  /**
+   * File to upload
+   */
+  file: MultipartFile;
+
+  /**
+   * File type
+   */
+  type: MultipartValue<FileTypes>;
+
+  /**
+   * Note public id, if file is related to a note
+   */
+  notePublicId?: MultipartValue<NotePublicId>;
+}
+
+const UploadRouter: FastifyPluginCallback<UploadRouterOptions> = (fastify, opts, done) => {
+  const { fileUploaderService } = opts;
+  const { noteService } = opts;
+
+  /**
+   * Prepare note id resolver middleware
+   * It should be used in routes that accepts note public id
+   */
+  const { softNoteResolverForUploadFileRequest } = useNoteResolver(noteService);
+
+  fastify.post<{
+    Body: UploadFileBody;
```

Comment:
this type is used only in a single place, so you can declare it in-place instead of moving to a separate interface. It will increase readability

Reply:
moved